From 853e2085d26cc4efeeb84ac5081c67d4c9369133 Mon Sep 17 00:00:00 2001
From: Zhuang Jin Can <jin.can.zhuang@intel.com>
Date: Tue, 17 Mar 2015 14:55:28 +0800
Subject: [PATCH] xhci: set ssic port unused only if xhci_suspend succeeds

If xhci_suspend() fails, we should not set port unused. Otherwise,
the port is set to unused without further entering D3, and the
port not usable anymore.

Change-Id: I947a2570270683da53a5de276408db92d4b9163f
Signed-off-by: Zhuang Jin Can <jin.can.zhuang@intel.com>
---
 drivers/usb/host/xhci-pci.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/host/xhci-pci.c b/drivers/usb/host/xhci-pci.c
index 42d5129..c48a481 100644
--- a/drivers/usb/host/xhci-pci.c
+++ b/drivers/usb/host/xhci-pci.c
@@ -301,7 +301,7 @@ static int xhci_pci_suspend(struct usb_hcd *hcd, bool do_wakeup)
 	/* This is SW workaround for spurious PME issue and HCRST hang problem
 	 * It required to set anc clear SSIC_PORT_UNUSED bit in D3 entry and
 	 * D3 exit. */
-	if (xhci->quirks & XHCI_SPURIOUS_PME)
+	if (!retval && xhci->quirks & XHCI_SPURIOUS_PME)
 		xhci_intel_ssic_port_unused(xhci, 1);
 
 	return retval;
-- 
1.9.1

